### ISSUE TITLE
Feature Request : Mongo connection string method is deprecated. Option to use new connection method.

### ISSUE THREAD

**Issue by dev-fullstacktechnology** [typeorm/typeorm#2270]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
There is no necessary steps to reproduce warning. Just create any database connection using sample mongodb and it will come back.
```
(node:27508) DeprecationWarning: current URL string parser is deprecated, and will be removed in a future version. To use the new parser, pass option { useNewUrlParser: true } to MongoClient.connect.
```

**Comment by pleerock** [comment]
What mongodb driver version are you using?

**Comment by dev-fullstacktechnology** [comment]
I am using `"mongodb": "^3.1.0-beta4"` in my project.

**Comment by pleerock** [comment]
So according to [this SO answer](https://stackoverflow.com/questions/50448272/avoid-current-url-string-parser-is-deprecated-warning-by-setting-usenewurlpars) you either shall not use beta yet either to explicitly provide a port in connection options. If its something ORM builds for you then try to find this place in MongodbDriver.ts and provide a port explicitly in the case if port was not specified in connection options and sent a PR with this fix, I'll merge it.

**Comment by dev-fullstacktechnology** [comment]
Here is my connection options. I have defined all necessary properties to connect database and still its giving me warning.

**Comment by pleerock** [comment]
can you please research what exactly gives you this error?

**Comment by mcMickJuice** [comment]
From the SO posted above, [this response](https://stackoverflow.com/a/50460871/1594351) has the information you need.

TL;DR; pin your version of mongodb to ~3.0.8 (or really, ~3.0.x).  I did this in my project and these warnings went away.  This is unrelated to current TypeORM connection settings as far as I can tell.

**Comment by galkin** [comment]
@pleerock, I opened pull with the fix for this issue. Could you review?

**Comment by pleerock** [comment]
fixed by #2772
